core/schema/src/main/java/org/hawkular/metrics/schema/SchemaService.java

```diff
@@ -50,56 +51,91 @@ public void run(Session session, String keyspace, boolean resetDB, int replicati
     public void run(Session session, String keyspace, boolean resetDB, int replicationFactor, boolean updateVersion) {
         CassalogBuilder builder = new CassalogBuilder();
         Cassalog cassalog = builder.withKeyspace(keyspace).withSession(session).build();
-        Map<String, ?> vars  = ImmutableMap.of(
+        URI script = getScript();
+        Map<String, ?> vars = this.getVars(session, keyspace, resetDB, replicationFactor);
+        cassalog.execute(script, this.getTags(), vars);
+
+        if (updateVersion) {
+            updateVersion(session, keyspace, 0, 0);
+        }
+    }
+
+    private Map<String, ?> getVars(Session session, String keyspace, boolean resetDB, int replicationFactor) {
+        Map<String, ?> vars = ImmutableMap.of(
                 "keyspace", keyspace,
                 "reset", resetDB,
                 "session", session,
                 "replicationFactor", replicationFactor,
                 "logger", logger
         );
+        return vars;
+    }
+
+    private List<String> getTags() {
         // TODO Add logic to determine the version tags we need to pass
         // For now, I am just hard coding the version tag, but a more robust solution would be
         // to calculate the tags using the current version stored in the sys_config table
         // and the new version which we can extract from any of our JAR manifest files.
         List<String> tags = asList("0.15.x", "0.18.x", "0.19.x", "0.20.x", "0.21.x", "0.23.x", "0.26.x", "0.27.x",
                 "0.30.x");
-        URI script = getScript();
-        cassalog.execute(script, tags, vars);
-
-        if (updateVersion) {
-            updateVersion(session, keyspace, 0, 0);
-        }
+        return tags;
     }
 
     public void updateVersion(Session session, String keyspace, long delay, int maxRetries) {
         doVersionUpdate(session, keyspace, delay, maxRetries);
     }
 
+    public void updateSchemaVersionSession(Session session, String keyspace, long delay, int maxRetries) {
+        doSchemaVersionUpdate(session, keyspace, delay, maxRetries);
+    }
+
+    public String getSchemaVersion(Session session, String keyspace) {
+        URI script = getScript();
+        CassalogBuilder builder = new CassalogBuilder();
+        Cassalog cassalog = builder.withKeyspace(keyspace).withSession(session).build();
+        Map<String, ?> vars = this.getVars(session, keyspace, false, 1);
+        List<ChangeSet> changeSets = cassalog.load(script, this.getTags(), vars);
+        if (changeSets.size() > 0) {
+            return changeSets.get(changeSets.size() - 1).getVersion();
+        }
+        return "";
```

Comment:
I feel like returning an empty string here is akin to swallowing an exception. In other words, ```changeSets``` should not be empty. If it is, then something is wrong, and I think we ought to throw an exception.

Reply:
+1, I can throw an exception and log an error here, just in case.